### PR TITLE
GLUE for non member functions

### PR DESCRIPTION
**Pull request by gicmo**
For Visual Studio 12

### CODE REVIEW

**Comment by asobolev** [comment]
Vanilla is super!

Another part of the C++ art)

But do we gain anything from having "describe" working via glue? IMHO this only brings complexity to the whole solution

**Comment by gicmo** [comment]
I want to get rid of the whole old part of how we dispatched functions, because we currently have two ways to do it (one via hash-maps, one via loop and compare). I did it the not-complex way, btw, because used a new function `desc` and didn't specialise the reg function (which would be possible too).

File: src/utils/glue.h [hunk 1]
```diff
@@ -211,6 +217,21 @@ struct funcbox : box {
     matryoshka_t args;
 };
 
+template<typename Klazz>
+struct getter : box {
```
Comment (asobolev) [inline src/utils/glue.h]: maybe call it different than "getter" as there are so many getters everywhere) maybe smth like "applicable" or "cast_and_apply" as it mainly casts the object from the first argument and applies the given function to it

Comment (gicmo) [inline src/utils/glue.h]: Names are hard. `entity_return_array_box`? Without the box? ;-)

Comment (asobolev) [inline src/utils/glue.h]: maybe) without the box)) but really, that's a minor thing. We could have a getter too. Just make it the way you like better and I merge it.

**Comment by gicmo** [comment]
The alternative would be to have one other function instead of `.desc` that takes a function pointer in the form of our old `fn_t` and not special case the `.desc` like case. Or have both.

**Pull request merged**